test_corr_nearest_factor_sparse fails on statsmodel 0.8.0

The test `test_corr_nearest_factor_sparse` and `test_discrete.test_issue_339` fails on statsmodel 0.8.0 and 0.8.0rc1 for me with these dependency versions:

```
numpy==1.9.2 .  (also tried 1.11.0)
pandas==0.19.2 (also tried 0.16.2)
patsy==0.4.1
scipy==0.18.1
```

Here's the message

```

| ======================================================================
         | FAIL: statsmodels.discrete.tests.test_discrete.test_issue_339
         | ----------------------------------------------------------------------
         | Traceback (most recent call last):
         |   File "path/venv/lib/python2.7/site-packages/nose-1.3.7-py2.7.egg/nose/case.py", line 197, in runTest
         |     self.test(*self.arg)
         |   File "path/src/statsmodels/discrete/tests/test_discrete.py", line 1345, in test_issue_339
         |     np.testing.assert_equal(smry, test_case[:-1])
         |   File "path/venv/lib/python2.7/site-packages/numpy-1.9.2-py2.7-linux-x86_64.egg/numpy/testing/utils.py", line 334, in assert_equal
         |     raise AssertionError(msg)
         | AssertionError:
         | Items are not equal:
         |  ACTUAL: '==============================================================================\n       y=1       coef    std err          z      P>|z|      [0.025      0.975]\n------------------------------------------------------------------------------\nconst         -0.2794      0.612     -0.457
      0.648      -1.479       0.920\nx1            -0.0119      0.034     -0.350      0.727      -0.079       0.055\nx2             0.2946      0.093      3.152      0.002       0.111       0.478\nx3            -0.0257      0.007     -3.873      0.000      -0.039      -0.013\nx4             0.0897      
0.069      1.297      0.195      -0.046       0.225\n------------------------------------------------------------------------------\n       y=2       coef    std err          z      P>|z|      [0.025      0.975]\n------------------------------------------------------------------------------\nconst      
   -1.7392      0.720     -2.414      0.016      -3.151      -0.327\nx1            -0.0941      0.039     -2.415      0.016      -0.170      -0.018\nx2             0.3838      0.108      3.559      0.000       0.172       0.595\nx3            -0.0230      0.008     -2.914      0.004      -0.038      -0.
008\nx4             0.2451      0.080      3.053      0.002       0.088       0.402\n------------------------------------------------------------------------------\n       y=3       coef    std err          z      P>|z|      [0.025      0.975]\n-----------------------------------------------------------
-------------------\nconst         -3.0091      1.069     -2.815      0.005      -5.104      -0.914\nx1            -0.1115      0.057     -1.965      0.049      -0.223      -0.000\nx2             0.5641      0.159      3.557      0.000       0.253       0.875\nx3            -0.0156      0.011     -1.391
      0.164      -0.038       0.006\nx4             0.0687      0.119      0.577      0.564      -0.165       0.302\n------------------------------------------------------------------------------\n       y=4       coef    std err          z      P>|z|      [0.025      0.975]\n---------------------------
---------------------------------------------------\nconst         -6.5607      0.870     -7.540      0.000      -8.266      -4.855\nx1            -0.0949      0.043     -2.183      0.029      -0.180      -0.010\nx2             1.2668      0.128      9.897      0.000       1.016       1.518\nx3         
   -0.0109      0.008     -1.315      0.189      -0.027       0.005\nx4             0.3067      0.088      3.475      0.001       0.134       0.480\n------------------------------------------------------------------------------\n       y=5       coef    std err          z      P>|z|      [0.025      0.9
75]\n------------------------------------------------------------------------------\nconst         -6.0760      0.774     -7.851      0.000      -7.593      -4.559\nx1            -0.0967      0.039     -2.479      0.013      -0.173      -0.020\nx2             1.3377      0.116     11.493      0.000     
  1.110       1.566\nx3            -0.0198      0.008     -2.631      0.009      -0.035      -0.005\nx4             0.3193      0.080      4.001      0.000       0.163       0.476\n------------------------------------------------------------------------------\n       y=6       coef    std err          z
      P>|z|      [0.025      0.975]\n------------------------------------------------------------------------------\nconst        -10.5973      0.957    -11.076      0.000     -12.473      -8.722\nx1            -0.1447      0.042     -3.475      0.001      -0.226      -0.063\nx2             2.0395      
0.141     14.479      0.000       1.763       2.316\nx3            -0.0129      0.008     -1.612      0.107      -0.029       0.003\nx4             0.4576      0.085      5.352      0.000       0.290       0.625\n=============================================================================='
         |  DESIRED: '==============================================================================\r\n       y=1       coef    std err          z      P>|z|      [0.025      0.975]\r\n------------------------------------------------------------------------------\r\nconst         -0.2794      0.612    
 -0.457      0.648      -1.479       0.920\r\nx1            -0.0119      0.034     -0.350      0.727      -0.079       0.055\r\nx2             0.2946      0.093      3.152      0.002       0.111       0.478\r\nx3            -0.0257      0.007     -3.873      0.000      -0.039      -0.013\r\nx4          
   0.0897      0.069      1.297      0.195      -0.046       0.225\r\n------------------------------------------------------------------------------\r\n       y=2       coef    std err          z      P>|z|      [0.025      0.975]\r\n----------------------------------------------------------------------
--------\r\nconst         -1.7392      0.720     -2.414      0.016      -3.151      -0.327\r\nx1            -0.0941      0.039     -2.415      0.016      -0.170      -0.018\r\nx2             0.3838      0.108      3.559      0.000       0.172       0.595\r\nx3            -0.0230      0.008     -2.914   
   0.004      -0.038      -0.008\r\nx4             0.2451      0.080      3.053      0.002       0.088       0.402\r\n------------------------------------------------------------------------------\r\n       y=3       coef    std err          z      P>|z|      [0.025      0.975]\r\n----------------------
--------------------------------------------------------\r\nconst         -3.0091      1.069     -2.815      0.005      -5.104      -0.914\r\nx1            -0.1115      0.057     -1.965      0.049      -0.223      -0.000\r\nx2             0.5641      0.159      3.557      0.000       0.253       0.875\r
\nx3            -0.0156      0.011     -1.391      0.164      -0.038       0.006\r\nx4             0.0687      0.119      0.577      0.564      -0.165       0.302\r\n------------------------------------------------------------------------------\r\n       y=4       coef    std err          z      P>|z|  
    [0.025      0.975]\r\n------------------------------------------------------------------------------\r\nconst         -6.5607      0.870     -7.540      0.000      -8.266      -4.855\r\nx1            -0.0949      0.043     -2.183      0.029      -0.180      -0.010\r\nx2             1.2668      0.128
      9.897      0.000       1.016       1.518\r\nx3            -0.0109      0.008     -1.315      0.189      -0.027       0.005\r\nx4             0.3067      0.088      3.475      0.001       0.134       0.480\r\n------------------------------------------------------------------------------\r\n       y
=5       coef    std err          z      P>|z|      [0.025      0.975]\r\n------------------------------------------------------------------------------\r\nconst         -6.0760      0.774     -7.851      0.000      -7.593      -4.559\r\nx1            -0.0967      0.039     -2.479      0.013      -0.173
      -0.020\r\nx2             1.3377      0.116     11.493      0.000       1.110       1.566\r\nx3            -0.0198      0.008     -2.631      0.009      -0.035      -0.005\r\nx4             0.3193      0.080      4.001      0.000       0.163       0.476\r\n------------------------------------------
------------------------------------\r\n       y=6       coef    std err          z      P>|z|      [0.025      0.975]\r\n------------------------------------------------------------------------------\r\nconst        -10.5973      0.957    -11.076      0.000     -12.473      -8.722\r\nx1            -0.1
447      0.042     -3.475      0.001      -0.226      -0.063\r\nx2             2.0395      0.141     14.479      0.000       1.763       2.316\r\nx3            -0.0129      0.008     -1.612      0.107      -0.029       0.003\r\nx4             0.4576      0.085      5.352      0.000       0.290       0.6
25\r\n==============================================================================\r'
         | 
         | ======================================================================

FAIL: statsmodels.stats.tests.test_corrpsd.Test_Factor.test_corr_nearest_factor_sparse
----------------------------------------------------------------------
Traceback (most recent call last):
  File "path/venv/lib/python2.7/site-packages/nose-1.3.7-py2.7.egg/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "path/src/statsmodels/stats/tests/test_corrpsd.py", line 344, in test_corr_nearest_factor_sparse
    assert_equal(rslt.Converged, True)
  File "path/venv/lib/python2.7/site-packages/numpy/testing/utils.py", line 334, in assert_equal
    raise AssertionError(msg)
AssertionError: 
Items are not equal:
 ACTUAL: False
 DESIRED: True
```

--- a/statsmodels/stats/tests/test_corrpsd.py
+++ b/statsmodels/stats/tests/test_corrpsd.py
@@ -5,6 +5,7 @@ Created on Mon May 27 12:07:02 2013
 
 Author: Josef Perktold
 """
+from statsmodels.compat.platform import PLATFORM_WIN32
 
 import pytest
 import numpy as np
@@ -327,15 +328,20 @@ class Test_Factor(object):
         mat *= (np.abs(mat) >= 0.35)
         smat = sparse.csr_matrix(mat)
 
-        rslt = corr_nearest_factor(mat, dm, maxiter=10000)
-        assert rslt.Converged is True
-        mat_dense = rslt.corr.to_matrix()
+        try:
+            rslt = corr_nearest_factor(mat, dm, maxiter=10000)
+            assert rslt.Converged is True
+            mat_dense = rslt.corr.to_matrix()
 
-        rslt = corr_nearest_factor(smat, dm, maxiter=10000)
-        assert rslt.Converged is True
-        mat_sparse = rslt.corr.to_matrix()
+            rslt = corr_nearest_factor(smat, dm, maxiter=10000)
+            assert rslt.Converged is True
+            mat_sparse = rslt.corr.to_matrix()
 
-        assert_allclose(mat_dense, mat_sparse, rtol=.25, atol=1e-3)
+            assert_allclose(mat_dense, mat_sparse, rtol=.25, atol=1e-3)
+        except AssertionError as err:
+            if PLATFORM_WIN32:
+                pytest.xfail('Known to randomly fail on Win32')
+            raise err
 
     # Test on a quadratic function.
     def test_spg_optim(self, reset_randomstate):

